base/iterators.jl

```diff
@@ -1070,6 +1124,7 @@ end
 function fixpoint_iter_type(itrT::Type, valT::Type, stateT::Type)
     nextvalstate = Base._return_type(next, Tuple{itrT, stateT})
     nextvalstate <: Tuple{Any, Any} || return Any
+    nextvalstate === Union{} && return Union{}
```

Comment:
Unrelated?

Reply:
Mostly? Prevents things from erroring in the wrong place if you mess up how to do iteration.